src/txn/txn.c

```diff
@@ -714,8 +858,18 @@ __txn_resolve_prepared_op(WT_SESSION_IMPL *session, WT_TXN_OP *op, bool commit,
          * Resolve the prepared update to be committed update.
          */
         __txn_resolve_prepared_update(session, upd);
+        resolved = true;
     }
 
+    /*
+     * Fix the history store contents if they exist. Even when there are no more updates in the
+     * update list, it is possible that there may be some older updates that are moved into history
+     * store when the page gets evicted.
+     */
+    if (!F_ISSET(S2C(session), WT_CONN_IN_MEMORY) && resolved &&
+      (upd == NULL || F_ISSET(upd, WT_UPDATE_HS)))
+        __txn_fixup_prepared_update(session, op, *cursorp, commit);
+
```

Comment:
The return of __txn_fixup_prepared_update is ignored?

Reply:
Fixed.